lib/vnode-store.js

```diff
@@ -87,28 +99,36 @@ VNodeStore.prototype.loadVNodeKeys = function loadVNodeKeys(vnodeName, forEachKe
  * wants to call on each key.  If the forEachKey returns false, we will remove the
  * key from bookkeeping.
  * @param {Error} err Error object if any persistence issues when fetching the set.
+ * @param {string} vnodeName The name of the vnode who owns the keys.
  * @param {Array} keys The entire array of keys loaded from memory.
  * @param {function} forEachKey The method to be called onthe recovery of each key.
+ * @param {function} done The method to call when all keys have been handled.
  */
-VNodeStore.prototype.onKeysLoaded = function onKeysLoaded(err, keys, forEachKey) {
+VNodeStore.prototype.onKeysLoaded = function onKeysLoaded(err, vnodeName, keys, forEachKey, done) {
+    var self = this;
     var nodeKeyArray = keys || [];
     var keysToRemove = [];
     if (typeof forEachKey === 'function') {
-        for (var i=0; i<vnodeKeyArray.length; i++) {
-            var removeKey = forEachKey(vnodeKeyArray[i]); 
-            if (removeKey) {
-                keysToRemove.append(vnodeKeyArray[i]);
-            }
-        }
+        async.filter(vnodeKeyArray, forEachKey, function(completedKeys) {
```

Comment:
I don't really like async.filter because it has a weird interface for the callback that doesn't return errors, just true/false.